integrations/langchain/tests/test_toolkit.py

```diff
@@ -0,0 +1,160 @@
+from unitycatalog_ai_langchain.toolkit import LangchainToolkit
+import json
+from contextlib import contextmanager
+from typing import NamedTuple
+import logging
+import uuid
+from unittest import mock
+from databricks.sdk.service.catalog import (
+    FunctionInfo,
+    FunctionParameterInfo,
+    FunctionParameterInfos,
+)
+
+
+@pytest.fixture
+def client() -> DatabricksFunctionClient:
+    with mock.patch(
+        "unitycatalog.ai.databricks.get_default_databricks_workspace_client",
+        return_value=mock.Mock(),
+    ):
+        return DatabricksFunctionClient(warehouse_id="warehouse_id", cluster_id="cluster_id")
+
+
+CATALOG = "ml"
+SCHEMA = "serena_uc_test"
+_logger = logging.getLogger(__name__)
+
+
+class FunctionObj(NamedTuple):
+    full_function_name: str
+    comment: str
+
+
+@contextmanager
+def create_function_and_cleanup(client: DatabricksFunctionClient):
+    func_name = f"{CATALOG}.{SCHEMA}.test_{uuid.uuid4().hex[:4]}"
+    comment = "Executes Python code and returns its stdout."
+    sql_body = f"""CREATE OR REPLACE FUNCTION {func_name}(code STRING COMMENT 'Python code to execute. Remember to print the final result to stdout.')
+RETURNS STRING
+LANGUAGE PYTHON
+COMMENT '{comment}'
+AS $$
+    import sys
+    from io import StringIO
+    stdout = StringIO()
+    sys.stdout = stdout
+    exec(code)
+    return stdout.getvalue()
+$$
+"""
+    try:
+        client.create_function(sql_function_body=sql_body)
+        yield FunctionObj(full_function_name=func_name, comment=comment)
+    finally:
+        try:
+            client.client.functions.delete(func_name)
+        except Exception as e:
+            _logger.warning(f"Fail to delete function: {e}")
+
+
+@pytest.fixture
+def set_default_client(client: DatabricksFunctionClient):
+    set_uc_function_client(client)
+    yield
+    set_uc_function_client(None)
+
+
+@pytest.mark.skipif(
+    os.environ.get("TEST_IN_DATABRICKS", "false").lower() != "true",
+    reason="This function test relies on connecting to a databricks workspace",
+)
+def test_toolkit_e2e(set_default_client):
+    client = get_uc_function_client()
+    with create_function_and_cleanup(client) as func_obj:
+        toolkit = LangchainToolkit(function_names=[func_obj.full_function_name])
+        tools = toolkit.tools
+        assert len(tools) == 1
+        tool = tools[0]
+        assert tool.name == func_obj.full_function_name
+        assert tool.description == func_obj.comment
+        tool.args_schema(**{"code": "print(1)"})
+        result = json.loads(tool.func(code="print(1)"))["value"]
+        assert result == "1\n"
+
+        toolkit = LangchainToolkit(function_names=[f"{CATALOG}.{SCHEMA}.*"])
+        assert len(toolkit.tools) >= 1
+        assert func_obj.full_function_name in [t.name for t in toolkit.tools]
+
+
+@pytest.mark.skipif(
+    os.environ.get("TEST_IN_DATABRICKS", "false").lower() != "true",
+    reason="This function test relies on connecting to a databricks workspace",
+)
```

Comment:
It looks like this marker is used a lot. Can we create a decorator that looks like:

```python
def requires_databricks(test_func):
    return pytest.mark.skipif(
        os.environ.get("TEST_IN_DATABRICKS", "false").lower() != "true",
        reason="This function test relies on connecting to a databricks workspace",
    )(test_func)
```